src/main/java/io/cdap/e2e/pages/actions/CdfStudioActions.java

```diff
@@ -498,11 +505,35 @@ public static void closeStatusBanner() {
 
   /**
    * Click on the element and move.
+   *
    * @param pluginName plugin title.
-   * @param xOffset horizontal move offset.
-   * @param yOffset vertical move offset.
+   * @param xOffset    horizontal move offset.
+   * @param yOffset    vertical move offset.
    */
   public static void movePlugin(String pluginName, int xOffset, int yOffset) {
     ElementHelper.dragAndDropByOffset(CdfStudioLocators.locatePluginNodeInCanvas(pluginName), xOffset, yOffset);
   }
+
+  /**
+   * Click on the Upload Button
+   */
+  public static void clickOnAddEntityButton() {
+    ElementHelper.clickOnElement(CdfStudioLocators.uploadFileButton);
+  }
+
+  public static void clickOnFixAllButton() {
+    ElementHelper.clickIfDisplayed(CdfStudioLocators.locatorOfFixAll());
+  }
+
+  public static void importPipeline(String filePath) throws URISyntaxException {
+    WaitHelper.waitForElementToBeDisplayed(CdfStudioLocators.importPipelineButton, 30);
+    FileImportUtil.uploadFile(CdfStudioLocators.importPipelineInputTag(), filePath);
+  }
+
+  public static void renameThePipeline() {
```

Comment:
There is already a fillPipelineNameAndSave action, why not use that?

Reply:
When importing the pipeline json and adding a suffix for renaming the pipeline. 'renameThePipeline' is used for already named pipelines.

Reply:
Logically, from clickOnElement --> fillPipelineNameAndSave are exactly the same steps that are performed in saveThePipeline step and fillPipelineNameAndSave method. Besides, this action doesn't work when used in the wrangler PR either as it timeout at this step. The error in the code is because:
1. pipeline name element is selected
2. pipeline name is generated
3. pipeline name element is selected again (inside fillPipelineNameAndSave call)

Either fix the method or if the intended action is the same, use the existing saveAndDeployPipeline step in the other PR to avoid confusion

Reply:
Resolved.